src/Common/Spectra/spherical_helper.jl

```diff
@@ -0,0 +1,307 @@
+
+    var_fourier = zeros(Complex{Float64}, nλ, nθ, nd)
+    var_grid = zeros(Float64, nλ, nθ, nd)
+    var_spherical = zeros(Complex{Float64}, num_fourier + 1, num_spherical + 1, nd)
+    var_spectrum = zeros(Float64, num_fourier + 1, num_spherical + 1, nd)
+
+    SpectralSphericalMesh( num_fourier, num_spherical, nλ, nθ, nd, Δλ, qwg, qnm, wave_numbers,
+    var_grid, var_fourier, var_spherical, var_spectrum )
+end
+
+"""
+    ComputeLegendre!(num_fourier, num_spherical, sinθ, nθ)
+
+Normalized associated Legendre polynomials
+
+  f(θ, λ) = ∑_{l=0} ∑_{m=-l}^{l} f_{lm} P_{lm}(sinθ)e^{imλ} (Y_{l,m} = P_{l,m} e^{i m λ} )
+  l=0,1...∞    and m = -l, -l+1, ... l-1, l
+  P_{0,0} = 1, such that 1/4π ∫∫YYdS = δ
+  P_{m,m} = sqrt((2m+1)/2m) cosθ P_{m-1m-1}
+  P_{m+1,m} = sqrt(2m+3) sinθ P_{m m}
+  sqrt((l^2-m^2)/(4l^2-1))P_{l,m} = x  P_{l-1, m} -  sqrt(((l-1)^2-m^2)/(4(l-1)^2 - 1))P_{l-2,m}
+  ε[m,l] = sqrt((l^2- m^2)/(4l^2 - 1))
+  (1-μ^2)d P_{m,l}/dμ = -nε[m,l+1]P_{m,l+1} + (l+1)ε_{m,l}P_{m,l-1}
+
+  Julia index starts with 1, so qnm[m+1,l+1] = P_l^m
+
+# Arguments:
+- num_fourier
+- num_spherical
+- sinθ
+- nθ
+
+# References:
+- Ehrendorfer, M., Spectral Numerical Weather Prediction Models, Appendix B, Society for Industrial and Applied Mathematics, 2011
+- Winch, D., Spherical harmonics, in Encyclopedia of Geomagnetism and Paleomagnetism, Eds Gubbins D. and Herrero-Bervera, E., Springer, 2007
+"""
+function ComputeLegendre!(num_fourier, num_spherical, sinθ, nθ)
+    qnm = zeros(Float64, num_fourier+1, num_spherical+2, nθ)
+    dqnm = zeros(Float64, num_fourier+1, num_spherical+1, nθ)
+
+    cosθ = sqrt.(1 .- sinθ.^2)
+    ε = zeros(Float64, num_fourier+1, num_spherical+2)
+
+    qnm[1, 1, :] .= 1.0
+    for m = 1:num_fourier
+        qnm[m+1, m+1, :] = - sqrt((2m+1)/(2m)) .* cosθ .* qnm[m, m,:]
+        qnm[m, m+1, :] = sqrt(2*m+1) * sinθ .* qnm[m,m, :]
+    end
+    qnm[num_fourier+1, num_fourier+2, :] = sqrt(2*(num_fourier+2)) * sinθ .* qnm[num_fourier+1,num_fourier+1, :]
+
+    for m = 0:num_fourier
+        for l = m+2:num_spherical+1
+            ε1= sqrt(( (l-1) ^2 - m^2) ./ (4* (l-1)^2 - 1))
+            ε2 = sqrt((l^2 - m^2) ./ (4*l^2 - 1))
+            qnm[m+1,l+1,:] = (sinθ .* qnm[m+1,l,:] -  ε1*qnm[m+1,l-1,:])/ε2
+        end
+    end
+
+    return qnm[:,1:num_spherical+1,:]
+end
+
+
+"""
+    ComputeGaussian!(n)
+
+Compute sin(latitude) and the weight factors for Gaussian integration
+
+  Pn(x) is an odd function
+  solve half of the n roots and weightes of Pn(x) # n = 2n_half
+  P_{-1}(x) = 0
+  P_0(x) = 1
+  P_1(x) = x
+  nP_n(x) = (2n-1)xP_{n-1}(x) - (n-1)P_{n-2}(x)
+  P'_n(x) = n/(x^2-1)(xP_{n}(x) - P_{n-1}(x))
+  x -= P_n(x)/P'_{n}()
+  Initial guess xi^{0} = cos(π(i-0.25)/(n+0.5))
+  wi = 2/(1-xi^2)/P_n'(xi)^2
+
+# Arguments
+- n: number of latitudes
+
+# References
+- Ehrendorfer, M., Spectral Numerical Weather Prediction Models, Appendix B, Society for Industrial and Applied Mathematics, 2011
+
+"""
+function ComputeGaussian!(n)
```

Comment:
See the [coding style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines -- function names  should be lowercase with words separated by underscores as necessary to improve readability.

Reply:
Yep, I was planning to address the formatting once the functionality is tested by Jia and Daniel, but thanks for flagging this up! :)